astropy/modeling/projections.py

```diff
@@ -1983,8 +1990,10 @@ def evaluate(cls, x, y, matrix, translation):
 
     @staticmethod
     def _create_augmented_matrix(matrix, translation):
+        if hasattr(matrix, 'unit') and matrix.unit and hasattr(translation, 'unit') and translation.unit:
+            translation = translation.to(matrix.unit)
```

Comment:
I am not entirely sure this makes sense. I don't really know how the units in this matrix work.